src/ChunkedVestingVault.sol

```diff
@@ -92,4 +93,24 @@ contract ChunkedVestingVault is VestingVault, ChunkedVestingVaultArgs {
         }
         return (total, vestingPeriods());
     }
+
+    /**
+     * @notice helper function to get the next chunk index after the given amount
+     * @param amount The amount of tokens to get chunk index for
+     * @return total The total amount vested from last vest to the given index
+     * @return chunks The total number of chunks used so far
+     */
+    function getNextChunkForAmount(uint256 amount)
+        internal
+        view
+        returns (uint256 total, uint256 chunks)
+    {
+        for (uint256 i = vestedChunks; i < vestingPeriods(); i++) {
+            total += amountAtIndex(i);
+            if (total >= amount) {
```

Comment:
This in combination with line 69 `if (total != amount) revert InvalidClaim();` seems to imply that the user of this contract can only claim an amount that's equal to an integer amount of chunks. Is that right? Is that the functionality we need?

Reply:
Ok, looking further, it seems like a claim does so for all available funds, so this is fine. I'd probably think about if you shouldn't enforce `total != amount` here instead of later as well, but the correct functionality will go there here.

Reply:
yeah, you should only be able to claim an integer number of chunks - a chunk is either fully vested or not vested at all

Reply:
Good point about enforcing total == amount here.. I guess I'm concerned about _somehow_ it being slightly off (i.e. maybe rounding errors) and we accidentally brick the vault and can't withdraw

Reply:
Good point about enforcing total == amount here.. let me think about that a bit